### PR TITLE
[KOGITO-2854] drools-core unfork

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoSerializablePlaceholderResolverStrategy.java [hunk 1]
```diff
@@ -25,18 +25,16 @@
 import org.kie.api.marshalling.ObjectMarshallingStrategy;
 import org.kie.api.marshalling.ObjectMarshallingStrategyAcceptor;
 
-public class SerializablePlaceholderResolverStrategy
-    implements
-    ObjectMarshallingStrategy {
+public class KogitoSerializablePlaceholderResolverStrategy implements ObjectMarshallingStrategy {
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoSerializablePlaceholderResolverStrategy.java:28]: Do we need this rename?

File: drools/drools-core/src/main/java/org/drools/core/impl/EnvironmentFactory.java [hunk 2]
```diff
@@ -28,7 +28,7 @@ public static Environment newEnvironment() {
             Environment env = new EnvironmentImpl();
             env.set(EnvironmentName.OBJECT_MARSHALLING_STRATEGIES, 
                     new ObjectMarshallingStrategy [] {
-                        new SerializablePlaceholderResolverStrategy(ClassObjectMarshallingStrategyAcceptor.DEFAULT)
+                        new KogitoSerializablePlaceholderResolverStrategy(ClassObjectMarshallingStrategyAcceptor.DEFAULT)
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/impl/EnvironmentFactory.java:31]: I have the impression this is just a rename so we should be able to revert the change and delete this file ( to keep original `drools-core` version)

File: drools/drools-core/src/main/java/org/drools/core/impl/KogitoStatefulKnowledgeSessionImpl.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2010 Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/impl/KogitoStatefulKnowledgeSessionImpl.java:2]: 2020

File: drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoComponentFactoryFactory.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoComponentFactoryFactory.java:2]: 2020

File: drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoInternalFactHandle.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2005 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoInternalFactHandle.java:2]: 2020

File: drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoFactHandleFactory.java [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2010 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoFactHandleFactory.java:2]: 2020

File: drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoDefaultFactHandle.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/kogito/factory/KogitoDefaultFactHandle.java:2]: 2020

File: drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoMarshallerReaderContext.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+                marshalProcessInstances, marshalWorkItems, env);
+
+        if (this.kBase != null) {
+            this.kBase.getProcesses().forEach( p -> this.processes.put(p.getId(), p));
+        }
+    }
+
+    @Override
+    protected ObjectMarshallingStrategy[] getMarshallingStrategy() {
+        return new ObjectMarshallingStrategy[]{new KogitoSerializablePlaceholderResolverStrategy( ClassObjectMarshallingStrategyAcceptor.DEFAULT  )};
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoMarshallerReaderContext.java:86]: I have the impression `KogitoSerializablePlaceholderResolverStrategy` is just a rename and if we revert that, we should be able to remove this file too

File: drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoMarshallerWriteContext.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                                  ObjectMarshallingStrategyStore resolverStrategyFactory,
+                                  boolean marshalProcessInstances,
+                                  boolean marshalWorkItems,
+                                  Environment env) throws IOException {
+        super(stream, kBase, wm, sinks, resolverStrategyFactory, marshalProcessInstances, marshalWorkItems, env);
+    }
+
+    @Override
+    protected ObjectMarshallingStrategy[] getMarshallingStrategy() {
+        return new ObjectMarshallingStrategy[]{new KogitoSerializablePlaceholderResolverStrategy( ClassObjectMarshallingStrategyAcceptor.DEFAULT  )};
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoMarshallerWriteContext.java:58]: Same as above

File: drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoProcessMarshallerWriteContext.java [hunk 10]
```diff
@@ -30,7 +30,7 @@
  * Extension to default <code>MarshallerWriteContext</code> that allows to pass additional
  * information to marshaller strategies, such as process instance id, task it, state
  */
-public class ProcessMarshallerWriteContext extends MarshallerWriteContext {
+public class KogitoProcessMarshallerWriteContext extends MarshallerWriteContext {
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoProcessMarshallerWriteContext.java:33]: Do we need this rename?

File: drools/drools-core/src/main/java/org/drools/core/ruleunit/InternalStoreCallback.java [hunk 11]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2005 JBoss Inc
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/ruleunit/InternalStoreCallback.java:2]: This seems to be the only change so we can probably remove this file and use kie7 version

File: drools/drools-core/src/main/java/org/drools/core/time/KogitoTimerServiceFactory.java [hunk 12]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2010 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2005 JBoss Inc
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/time/KogitoTimerServiceFactory.java:2]: Back to the future

File: drools/drools-core/src/main/java/org/drools/core/time/KogitoTimerServiceFactory.java [hunk 13]
```diff
@@ -36,4 +37,12 @@ public static TimerService getTimerService(SessionConfiguration config) {
...
         return service;
     }
+
+    private static TimerService newTimerService(SessionConfigurationImpl config) {
+        String className = config.getPropertyValue( "drools.timerService", "org.drools.core.time.impl.JDKTimerService" );
+        if ( className == null ) {
+            return null;
+        }
+        return (TimerService) ComponentsFactory.createTimerService( className );
+    }
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/time/KogitoTimerServiceFactory.java:47]: I have the impression we can do this change in kie7 and get rid of this copy. Wdyt?

File: drools/drools-core/src/main/java/org/drools/core/util/KogitoStringUtils.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
+package org.drools.core.util;
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/util/KogitoStringUtils.java:1]: License

File: drools/drools-core/src/main/java/org/drools/core/util/KogitoStringUtils.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+
+public class KogitoStringUtils {
+
+    public static String capitalize(String string) {
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/util/KogitoStringUtils.java:5]: We can probably move this method to [StringUtils](https://github.com/kiegroup/drools/blob/master/drools-core/src/main/java/org/drools/core/util/StringUtils.java) class and remove this one. Btw that class already contains `ucFirst` method that is equivalment except for `null` check

File: drools/drools-core/src/main/resources/META-INF/kie.conf [hunk 16]
```diff
@@ -1,5 +1,2 @@
-org.kie.api.io.KieResources = org.drools.core.io.impl.ResourceFactoryServiceImpl
-org.kie.api.marshalling.KieMarshallers = org.drools.core.marshalling.impl.MarshallerProviderImpl
-org.kie.api.concurrent.KieExecutors = org.drools.core.concurrent.ExecutorProviderImpl
-
+org.drools.core.reteoo.KieComponentFactoryFactory=org.drools.core.kogito.factory.KogitoComponentFactoryFactory
```
Comment (danielezonca) [inline drools/drools-core/src/main/resources/META-INF/kie.conf:1]: Do we use `kie.conf` in Kogito?

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/EntryPointDataProcessor.java [hunk 17]
```diff
@@ -72,6 +73,6 @@ public void delete(DataHandle dh, RuleImpl rule, TerminalNode terminalNode, Fact
 
     public void delete(InternalFactHandle fh, RuleImpl rule, TerminalNode terminalNode, FactHandle.State fhState) {
         (( WorkingMemoryEntryPoint ) entryPoint).delete( fh, rule, terminalNode, fhState );
-        handles.remove( fh.getDataHandle() );
+        handles.remove( (( KogitoInternalFactHandle ) fh).getDataHandle() );
```
Comment (danielezonca) [inline drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/EntryPointDataProcessor.java:76]: This is a new class specific for `Kogito` so we should be able to update the signature to `KogitoInternalFactHandle` and avoid an internal cast.
Wdyt?

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/FieldDataStore.java [hunk 18]
```diff
@@ -92,14 +93,14 @@ public void subscribe(DataProcessor processor) {
...
         subscribers.forEach(s -> s.update(dh, (T) obj));
     }
 
     @Override
     public void delete(InternalFactHandle fh, RuleImpl rule, TerminalNode terminalNode, FactHandle.State fhState) {
-        DataHandle dh = fh.getDataHandle();
+        DataHandle dh = ((KogitoInternalFactHandle)fh).getDataHandle();
         if (dh != this.handle) {
             throw new IllegalArgumentException("The given handle is not contained in this DataStore");
         }
```
Comment (danielezonca) [inline drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/FieldDataStore.java:106]: Same as above

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/ListDataStore.java [hunk 19]
```diff
@@ -78,14 +79,14 @@ public void subscribe(DataProcessor processor) {
...
-        DataHandle dh = fh.getDataHandle();
+        DataHandle dh = ((KogitoInternalFactHandle)fh).getDataHandle();
         entryPointSubscribers.forEach( s -> s.update( dh, obj, mask, modifiedClass, activation ) );
         subscribers.forEach( s -> s.update(dh, (T) obj) );
     }
 
     @Override
     public void delete( InternalFactHandle fh, RuleImpl rule, TerminalNode terminalNode, FactHandle.State fhState) {
-        DataHandle dh = fh.getDataHandle();
+        DataHandle dh = ((KogitoInternalFactHandle)fh).getDataHandle();
```
Comment (danielezonca) [inline drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/ListDataStore.java:89]: Same as above

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/StringUtils.java [hunk 20]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2010 Red Hat, Inc. and/or its affiliates.
+ * Copyright 2005 JBoss Inc
```
Comment (danielezonca) [inline drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/StringUtils.java:2]: Back to the future

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/StringUtils.java [hunk 21]
```diff
@@ -14,17 +14,20 @@
  * limitations under the License.
  */
 
-package org.drools.core.util.asm;
+package org.kie.kogito.rules.units;
 
-public class MethodCompareB {
+public class StringUtils {
```
Comment (danielezonca) [inline drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/StringUtils.java:19]: To be removed? We already have `org.drools.core.util.KogitoStringUtils`

File: drools/kogito-scenario-simulation/README.md [hunk 22]
```diff
@@ -29,7 +29,7 @@ package testscenario;
  * KogitoJunitActivator is a custom JUnit runner that enables the execution of Test Scenario files (*.scesim).
  * This activator class, when executed, will load all scesim files available in the project and run them.
  * Each row of the scenario will generate a test JUnit result. */
-@org.junit.runner.RunWith(org.kogito.scenariosimulation.runner.KogitoJunitActivator.class)
+@org.junit.runner.RunWith(KogitoJunitActivator.class)
```
Comment (danielezonca) [inline drools/kogito-scenario-simulation/README.md:32]: Please revert this change in the `README`

File: jbpm/jbpm-flow/pom.xml [hunk 23]
```diff
@@ -36,6 +36,16 @@
...
+      <exclusions>
+        <exclusion>
+          <groupId>org.kie</groupId>
+          <artifactId>kie-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.kie</groupId>
+          <artifactId>kie-internal</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (danielezonca) [inline jbpm/jbpm-flow/pom.xml:48]: Is this needed? I expect that `org.kie.kogito:drools-core` already excludes them. If not, please fix it there so that users will not have to face with it

File: drools/drools-core-dynamic/pom.xml [hunk 24]
```diff
@@ -19,8 +19,18 @@
```
Comment (danielezonca) [inline drools/drools-core-dynamic/pom.xml:19]: Do we need `drools-core-dynamic` in Kogito? I expect only `-static` should be used

File: jbpm/jbpm-flow/src/main/java/org/jbpm/marshalling/impl/ProtobufProcessMarshaller.java [hunk 25]
```diff
@@ -293,17 +294,26 @@ public void writeWorkItem(MarshallerWriteContext context, org.drools.core.proces
...
 
+    @Override
+    public void writeProcessTimers(MarshallerWriteContext context) throws IOException {
+
+    }
+
+    @Override
+    public void readProcessTimers(MarshallerReaderContext context) throws IOException, ClassNotFoundException {
+
+    }
```
Comment (danielezonca) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/marshalling/impl/ProtobufProcessMarshaller.java:318]: Have we changed the interface? If these methods are not supposed to be used but just to comply the interface why don't raise an `UnsupportedOperationException`?

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightWorkItemManager.java [hunk 26]
```diff
@@ -218,14 +215,50 @@ public void registerWorkItemHandler(String workItemName, WorkItemHandler handler
...
+    public Set<WorkItem> getWorkItems() {
+        throw new UnsupportedOperationException( "org.jbpm.process.instance.LightWorkItemManager.getWorkItems -> TODO" );
+
+    }
+
+    @Override
+    public WorkItem getWorkItem( long id ) {
+        throw new UnsupportedOperationException( "org.jbpm.process.instance.LightWorkItemManager.getWorkItem -> TODO" );
+
+    }
```
Comment (danielezonca) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightWorkItemManager.java:246]: Do we have a ticket to track these TODOs?

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/event/DefaultSignalManager.java [hunk 27]
```diff
@@ -116,7 +117,7 @@ public SignalProcessInstanceAction(MarshallerReaderContext context) throws IOExc
 		}
 		
 		public void execute(InternalWorkingMemory workingMemory) {
-			ProcessInstance processInstance = workingMemory.getProcessInstance(processInstanceId);
+			ProcessInstance processInstance = ((KogitoWorkingMemory)workingMemory).getProcessInstance(processInstanceId);
```
Comment (danielezonca) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/event/DefaultSignalManager.java:120]: Is it possible to change the signature to use `KogitoWorkingMemory`?

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/impl/RuleConstraintEvaluator.java [hunk 28]
```diff
@@ -103,7 +104,7 @@ public boolean evaluate(NodeInstance instance,
         	((Node) instance.getNode()).getUniqueId() + "-" + 
         	((Node) connection.getTo()).getUniqueId() + "-" + connection.getToType();
 
-        return agenda.isRuleActiveInRuleFlowGroup( "DROOLS_SYSTEM", rule, processInstance.getId() );
+        return ((KogitoInternalAgenda)agenda).isRuleActiveInRuleFlowGroup("DROOLS_SYSTEM", rule, processInstance.getId() );
```
Comment (danielezonca) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/impl/RuleConstraintEvaluator.java:107]: Legacy code? Is RuleFlowGroup working in Kogito?

File: pom.xml [hunk 29]
```diff
@@ -141,6 +141,8 @@
     <version.org.mockito>3.3.3</version.org.mockito>
     <version.org.mvel>2.4.7.Final</version.org.mvel>
     <version.org.kie7>7.40.0.Final</version.org.kie7>
+<!--    <version.org.kie7>7.39.0.Final</version.org.kie7>-->
+    <version.org.kie7>7.41.0-SNAPSHOT</version.org.kie7>
```
Comment (danielezonca) [inline pom.xml:145]: To be updated before merging

File: drools/drools-core/src/main/java/org/drools/core/common/KogitoDefaultAgenda.java [hunk 30]
```diff
@@ -0,0 +1,36 @@
...
+    @Override
+    public void activateRuleFlowGroup(String name, String processInstanceId, String nodeInstanceId) {
+        InternalRuleFlowGroup ruleFlowGroup = (InternalRuleFlowGroup) getRuleFlowGroup( name );
+        activateRuleFlowGroup( ruleFlowGroup, processInstanceId, nodeInstanceId );
+    }
+
+    @Override
+    public boolean isRuleInstanceAgendaItem(String ruleflowGroupName, String ruleName, String processInstanceId) {
+        return isRuleInstanceAgendaItem(ruleflowGroupName, ruleName, (Object) processInstanceId);
+    }
```
Comment (danielezonca) [inline drools/drools-core/src/main/java/org/drools/core/common/KogitoDefaultAgenda.java:30]: These methods should be probably removed in `Kogito` because related to AgendaGroup/RuleFlowGroup

File: drools/drools-compiler/src/main/resources/META-INF/kie.conf [hunk 31]
```diff
@@ -1,2 +1 @@
-org.kie.api.KieServices = org.drools.compiler.kie.builder.impl.KieServicesImpl
-org.kie.internal.builder.KnowledgeBuilderFactoryService = org.drools.compiler.builder.impl.KnowledgeBuilderFactoryServiceImpl
+org.kie.internal.builder.KnowledgeBuilderFactoryService = org.drools.compiler.builder.impl.KogitoKnowledgeBuilderFactoryServiceImpl
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-compiler/src/main/resources/META-INF/kie.conf:1]: There are test in jbpm that needs to run with the dynamic classloader module. If they try to instance the drools KnowledgeBuilderFactoryService they fail because of the missing jaxb dependency required there.

File: drools/drools-compiler/src/test/java/org/drools/compiler/integrationtests/SerializationHelper.java [hunk 32]
```diff
@@ -43,36 +44,36 @@
...
 
-    public static ReadSessionResult getSerialisedStatefulKnowledgeSessionWithMessage(final KieSession ksession,
-                                                                                 final KieBase kbase,
-                                                                                 final boolean dispose) throws Exception {
-        final ProtobufMarshaller marshaller = (ProtobufMarshaller) MarshallerFactory.newMarshaller(kbase,
-                                                                                                   (ObjectMarshallingStrategy[]) ksession.getEnvironment().get(EnvironmentName.OBJECT_MARSHALLING_STRATEGIES));
+    public static ReadSessionResult getSerialisedStatefulKnowledgeSessionWithMessage( final KieSession ksession,
+                                                                                      final KieBase kbase,
+                                                                                      final boolean dispose) throws Exception {
+        ObjectMarshallingStrategy[] strategies = ( ObjectMarshallingStrategy[]) ksession.getEnvironment().get( EnvironmentName.OBJECT_MARSHALLING_STRATEGIES);
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-compiler/src/test/java/org/drools/compiler/integrationtests/SerializationHelper.java:72]: This is currently used only in jbpm tests. I'll move it there.

File: drools/drools-core/src/main/java/org/drools/core/audit/KogitoKnowledgeRuntimeLoggerProviderImpl.java [hunk 33]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2005 JBoss Inc
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/audit/KogitoKnowledgeRuntimeLoggerProviderImpl.java:2]: I'll check and fix all headers.

File: drools/drools-core/src/main/java/org/drools/core/audit/event/KogitoRuleFlowLogEvent.java [hunk 34]
```diff
@@ -16,16 +16,17 @@
...
  */
-public interface ILogEventFilter {
-
-    /**
-     * Returns whether the given event should be filtered from the event log or not.
-     * @param event The log event
-     * @return Whether the event should be filtered from the event log or not.
-     */
-    boolean acceptEvent(LogEvent event);
+public class KogitoRuleFlowLogEvent extends RuleFlowLogEvent {
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/audit/event/KogitoRuleFlowLogEvent.java:26]: There's a subtle problem here. Without this class that is explicitly calling `processInstance.getProcessId()` the same call will be made by the Drools class which will fail with a 

`java.lang.NoSuchMethodError: org.kie.api.runtime.process.ProcessInstance.getId()J`

because the kogito implementation of that method returns a String but drools expects a long.

File: drools/drools-core/src/main/java/org/drools/core/audit/event/KogitoRuleFlowNodeLogEvent.java [hunk 35]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.core.audit.event;
+
+import org.kie.api.runtime.process.ProcessInstance;
+
+public class KogitoRuleFlowNodeLogEvent extends RuleFlowNodeLogEvent {
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/audit/event/KogitoRuleFlowNodeLogEvent.java:21]: Same as above

File: drools/drools-core/src/main/java/org/drools/core/audit/event/KogitoRuleFlowVariableLogEvent.java [hunk 36]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.core.audit.event;
+
+import org.kie.api.runtime.process.ProcessInstance;
+
+public class KogitoRuleFlowVariableLogEvent extends RuleFlowVariableLogEvent {
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/audit/event/KogitoRuleFlowVariableLogEvent.java:21]: Same as above

File: drools/drools-core/src/main/java/org/drools/core/common/KogitoInternalAgenda.java [hunk 37]
```diff
@@ -0,0 +1,42 @@
...
+     */
+    void activateRuleFlowGroup(String name, String processInstanceId, String nodeInstanceId);
+
+    /**
+     * Returns true if there is at least one activation of the given rule name
+     * in the given ruleflow group name
+     */
+    boolean isRuleInstanceAgendaItem(String ruleflowGroupName,
+                                     String ruleName,
+                                     String processInstanceId);
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/common/KogitoInternalAgenda.java:41]: It is true that Kogito doesn't use RFG, but there are many tests in jbpm (inherited form v7) that still do and I wanted to keep everything working as before even though I agree that this could be no longer necessary.

File: drools/drools-core/src/main/java/org/drools/core/event/KogitoProcessNodeLeftEventImpl.java [hunk 38]
```diff
@@ -20,13 +20,13 @@
 import org.kie.api.runtime.KieRuntime;
 import org.kie.api.runtime.process.NodeInstance;
 
-public class ProcessNodeLeftEventImpl extends ProcessEvent implements ProcessNodeLeftEvent {
+public class KogitoProcessNodeLeftEventImpl extends ProcessEvent implements ProcessNodeLeftEvent {
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/event/KogitoProcessNodeLeftEventImpl.java:23]: See https://github.com/kiegroup/kogito-runtimes/pull/649#discussion_r460761656 If I don't do this the invocation of `nodeInstance.getId()` in the toString method of this event will cause a NoSuchMethodError.

File: drools/drools-core/src/main/java/org/drools/core/event/KogitoProcessNodeTriggeredEventImpl.java [hunk 39]
```diff
@@ -20,13 +20,13 @@
 import org.kie.api.runtime.KieRuntime;
 import org.kie.api.runtime.process.NodeInstance;
 
-public class ProcessNodeTriggeredEventImpl extends ProcessEvent implements ProcessNodeTriggeredEvent {
+public class KogitoProcessNodeTriggeredEventImpl extends ProcessEvent implements ProcessNodeTriggeredEvent {
```
[the remark this answers is not in the document]
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/event/KogitoProcessNodeTriggeredEventImpl.java:23]: Same as above

File: drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoSerializablePlaceholderResolverStrategy.java (hunk 1 above)
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoSerializablePlaceholderResolverStrategy.java:28]: The interface `ObjectMarshallingStrategy` in public API is indeed changed with that new dataType argument in the unmarshal method. The new implementation is necessary to implement this change which is used in a few tests like ProtoStreamObjectMarshallingStrategyTest

File: drools/drools-core/src/main/java/org/drools/core/impl/EnvironmentFactory.java (hunk 2 above)
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/impl/EnvironmentFactory.java:31]: See above

File: drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoMarshallerReaderContext.java (hunk 8 above)
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/marshalling/impl/KogitoMarshallerReaderContext.java:86]: As explained above this is necessary to implement the unmarshal method with the addition dataType argument.

File: drools/drools-core/src/main/resources/META-INF/kie.conf (hunk 16 above)
Comment (mariofusco) [reply drools/drools-core/src/main/resources/META-INF/kie.conf:1]: Some old test cases (maybe we should just remove them?) still use the dynamic class loader and then also this.

File: drools/kogito-scenario-simulation/README.md (hunk 22 above)
Comment (mariofusco) [reply drools/kogito-scenario-simulation/README.md:32]: Done.

File: drools/drools-core/src/main/java/org/drools/core/util/KogitoStringUtils.java (hunk 15 above)
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/util/KogitoStringUtils.java:5]: I removed this class and replaced all usages of ts capitalize method with ucFirst one.

File: drools/drools-core-dynamic/pom.xml (hunk 24 above)
Comment (mariofusco) [reply drools/drools-core-dynamic/pom.xml:19]: This is only used to run old test cases. I agree that it could be removed, but I'd leave it there for now.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/marshalling/impl/ProtobufProcessMarshaller.java (hunk 25 above)
Comment (mariofusco) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/marshalling/impl/ProtobufProcessMarshaller.java:318]: These methods shouldn't do anything (and actually not even exists) in Kogito. The problem is that they have to stay there to make the compiler happy and they cannot throw an Exception because they are still called by drools code (even when running in kogito) but are not supposed to do anything.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightWorkItemManager.java (hunk 26 above)
Comment (mariofusco) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/LightWorkItemManager.java:246]: They are not actual TODOs, it's only that my IDE generate methods to be implemented in that way. I removed the message inside the Exception.

File: drools/drools-core/src/main/java/org/drools/core/common/KogitoDefaultAgenda.java (hunk 30 above)
Comment (mariofusco) [reply drools/drools-core/src/main/java/org/drools/core/common/KogitoDefaultAgenda.java:30]: Agreed, as mentioned before I'm just keeping them for backward compatibility reasons and to run legacy tests.

File: jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/event/DefaultSignalManager.java (hunk 27 above)
Comment (mariofusco) [reply jbpm/jbpm-flow/src/main/java/org/jbpm/process/instance/event/DefaultSignalManager.java:120]: Unfortunately no, it has to implement the `PropagationEntry` interface.

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/EntryPointDataProcessor.java (hunk 17 above)
Comment (mariofusco) [reply drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/EntryPointDataProcessor.java:76]: Done.

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/FieldDataStore.java (hunk 18 above)
Comment (mariofusco) [reply drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/FieldDataStore.java:106]: Done.

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/ListDataStore.java (hunk 19 above)
Comment (mariofusco) [reply drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/ListDataStore.java:89]: Done.

File: jbpm/jbpm-flow/pom.xml (hunk 23 above)
Comment (mariofusco) [reply jbpm/jbpm-flow/pom.xml:48]: Removed.

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by mariofusco** [comment]
Jenkins test

**Comment by mariofusco** [comment]
jenkins retest this please

**Comment by mariofusco** [comment]
@danielezonca @evacchi @lucamolteni @cristianonicolai Can you please help me to give a look at this? Jenkins keeps reporting that there 3 failing tests, but I'm not able to reproduce those failures locally.

**Comment by lucamolteni** [comment]
jenkins retest this please

**Comment by lucamolteni** [comment]
@mariofusco on my machine all tests run, it seems like a maven/CI problem as the error is mostly
``` [ERROR] testProcessSignals  Time elapsed: 0.02 s  &lt;&lt;&lt; ERROR!
 java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for null
 Caused by: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for null
 Caused by: io.quarkus.bootstrap.resolver.maven.BootstrapMavenException: Failed to resolve artifacts
 Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: org.kie.   kogito:jbpm-flow-builder:jar:8.0.0-SNAPSHOT, org.kie.kogito:jbpm-bpmn2:jar:8.0.0-SNAPSHOT, org.kie.kogito:jbpm-serverless-      workflow:jar:8.0.0-SNAPSHOT, org.kie.kogito:kogito-codegen:jar:8.0.0-SNAPSHOT, org.kie.kogito:kogito-quarkus-deployment:jar:8.  0.0-SNAPSHOT: Could not find artifact org.kie.kogito:jbpm-flow-builder:jar:8.0.0-SNAPSHOT
 Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.kie.kogito:jbpm-flow-builder:jar: 8.0.0-SNAPSHOT```

I retriggered

**Comment by mariofusco** [comment]
> @mariofusco on my machine all tests run, it seems like a maven/CI problem as the error is mostly

Same things for me. How should we proceed? @cristianonicolai can you maybe give a look at this?

**Comment by evacchi** [comment]
hopefully this will become green as soon as https://github.com/kiegroup/kogito-runtimes/pull/673 is merged :crossed_fingers:

**Comment by mariofusco** [comment]
jenkins retest this please

**Comment by evacchi** [comment]
@mariofusco I think we have to create a branch on kogito-examples with this commit https://github.com/evacchi/kogito-examples/commit/116c8b2841f2b6ee21c2ed0099049e4995360af2

**Comment by lucamolteni** [comment]
@mariofusco yes @evacchi is correct remember I sent you the patch for the examples?
